engine/cluster/src/main/java/org/opennms/alec/engine/cluster/AbstractClusterEngine.java

```diff
@@ -111,11 +110,14 @@
 
     private boolean alarmsChangedSinceLastTick = false;
     private boolean feedbackChangedSinceLastTick = false;
-    private DijkstraShortestPath<CEVertex, CEEdge> shortestPath;
+    private final FilterableGraphManager graphManager =
+            FilterableGraphManager.withFilter(this::filterVertex);
+    private final SolvableGraph<CEVertex> shortestPath =
+            DijkstraSolvableGraph.newInstance(graphManager.getFilteredGraph(), e -> (double) e.getWeight(),
```

Comment:
Should we expose the thread pool size configuration or keep it internal?

Reply:
Since all the work being done is CPU bound/non-blocking, maybe it's better to just use the common pool for one less thing to have to configure. What do you think @j-white ?

Reply:
We can keep this internal for now. I would default to `Math.max(1, Runtime.getRuntime().availableProcessors() - 1)` so that we leave 1 core idle.